content/docs/user-guide/programming/autogen/_index.md

```diff
@@ -1,10 +1,237 @@
-title: Automating boilerplate with AZ Generator
+description: Use the Open 3D Engine (O3DE) Automated Generator to generate resources from Jinja2 templates. 
+linktitle: AzAutoGen source generator
+title: Automate source generation from templates with AzAutoGen
 weight: 200
 ---
 
-{{< todo issue="https://github.com/o3de/o3de.org/issues/551" >}}
-Write documentation for the AZ Generator system.
-{{< /todo >}}
+When confronted with the problem of many similar code objects or source inputs, the best solution for simple cases is often automated generation from templates and values. **Open 3D Engine (O3DE)** has its own lightweight generator system, *AzAutoGen*, to solve this problem. AzAutoGen uses the [Jinja](https://jinja.palletsprojects.com/) template engine and Python in combination with either XML or JSON input files to produce generated sets of output. Your own O3DE Gems and projects can take advantage of AzAutoGen during [build](/docs/user-guide/build/generated-source/) to generate source inputs.
+
+In this topic you'll be introduced to the conceptual idea behind AzAutoGen, and become familiar with both of the JSON and XML input formats. To learn more about Jinja templates, read the [Jinja template designer](https://jinja.palletsprojects.com/templates/) documentation.
+
+## How AzAutoGen works
+
+AzAutoGen works off of two input sets, a collection of value and template files, and a series of mappings from values to templates. AzAutoGen outputs to a specified file, or a set of files which contain special substitutions where When AzAutoGen is invoked, it runs through the following steps:
+
+1. Prune all input files without one of the extensions `.xml`, `.json`, or `.jinja`. Templates can be co-located with other code as a result, although this isn't recommended.
+1. Categorize the input files as source files (`.xml`, `.json`) and template files (`.jinja`).
+1. For each `input_map,template,output` tuple that makes up a rule, match the corresponding files into their appropriate sets. `template` must be a single file; Each input is run through this template to generate a corresponding file named according to the pattern in `output`. Each mapping argument allows for the following substitutions:
+
+   * Within `input_map`, the following matching operators are supported.
+     * `*` - Sequence of any length
+     * `?` - Single-character squence
+     * `[<sequence>]` - Matches any characters in `<sequence>`.
+     * `[!<sequence>]` - Matches any characters _not_ in `<sequence>`.
+   * Within `output`, the following tokens can be used to create unique file names for multiple inputs.
+      * `$path`  - The path to the final output destination.
+      * `$fileprefix` - The name of the current file up to the first `.` token. This is equivalent to running the UNIX shell command `basename ${file%%.*}`.
+      * `$file` - The full name of the input file. This is equivalent to running the UNIX shell command `basename $file`.
+
+   1. For each input file:
```

Comment:
Did you mean to indent this? It feels like it could be step 4 after the first 3 steps, but it's not because of the indentation.